core/arch/arm/kernel/thread.c

```diff
@@ -1521,8 +1521,13 @@ void thread_get_user_kdata(struct mobj **mobj, size_t *offset,
 
 	core_mmu_get_user_va_range(&v, NULL);
 	*va = v + thread_user_kcode_size;
-	*mobj = mobj_tee_ram;
+	*mobj = mobj_tee_ram_rw;
+#ifdef CFG_CORE_RWDATA_NOEXEC
+	*offset = (vaddr_t)thread_user_kdata_page - VCORE_START_VA -
```

Comment:
I believe it should be possible to calculate this offset based on the address returned by `mobj_get_va(mobj_tee_ram_rw, 0);`